### ISSUE TITLE
fix(html-spec): update optgroup content model for customizable select

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#3591]
## Describe the bug

The `permitted-contents` rule rejects `<div>` and `<legend>` as children of `<optgroup>`, but the customizable `<select>` spec allows these elements.

Related: #2069 (parent issue for customizable `<select>` support)

## Code Example

```html
<select>
  <optgroup label="Group">
    <div>
      <option>A</option>
    </div>
  </optgroup>
</select>

<select>
  <optgroup>
    <legend>Group</legend>
    <option>A</option>
  </optgroup>
</select>
```

## The current behavior

```
The text node is not allowed in the "optgroup" element in this context
```

## The expected behavior

No violations reported.

## Impact

Causes ~3 false positives in the nu-html-checker compatibility benchmark.

## Affected package

`@markuplint/html-spec` — optgroup element content model

**Comment by YusukeHirao** [comment]
This is tightly coupled with #2069 (customizable `<select>` support). The `optgroup` content model change (`legend` + inner content elements) is part of the same HTML spec update that adds `<button>`, `<div>`, `<selectedcontent>` etc. to `<select>`.

Updating `optgroup` alone without the full `<select>` content model update would be incomplete. Deferring to #2069.

**Comment by YusukeHirao** [comment]
Closing in favor of #2069 which covers the full customizable `<select>` content model update.

**Comment by YusukeHirao** [comment]
Reopening — closed prematurely based on the benchmark showing `match-error`, but that was markuplint's generic `permitted-contents` "unexpected text node" check firing for unrelated reasons, not the intended content-model fix. The actual `optgroup` content-model update (allowing `legend`/`noscript`/`div`) hasn't landed in `src/spec.optgroup.jsonc` yet. Still blocked on parse5 (`inikulin/parse5#1610`, still open as of 2026-08-30). Full status + concrete spec-data sketch posted on the parent issue: https://github.com/markuplint/markuplint/issues/2069#issuecomment-5467434192
